### ISSUE TITLE
Event driven scaler not scaling up

### ISSUE THREAD

**Issue by nosalan** [microsoft/azure-container-apps#913]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [X] bug report -> please search issues before submitting
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
Container Apps Environment with dedicated workload profile is not scaling up when a new job is submitted.
Configuration of environment workload profiles: 
- Dedicated (vCPUs:4, Memory 16 Gi), Current instances: 0, Min & Max number of instances: 0 - 4 (**scaling from zero to four nodes**)

Container App Job configuration:
![image](https://github.com/microsoft/azure-container-apps/assets/2273250/77a22bb7-a4d1-41b1-98b9-684695d42452)
Containers:
![image](https://github.com/microsoft/azure-container-apps/assets/2273250/85e5e8d6-69f4-4838-8f6f-7dc6eafa350c)
Event Driver scaling:
![image](https://github.com/microsoft/azure-container-apps/assets/2273250/4ec72a3b-a8bc-472a-abdb-9bfd6d7fcda9)

### Steps to reproduce
1.Configure CA Environment and Job like above.
2.Submit a job, here event driven scaling is used to run a job when pipeline is executed [like here](https://learn.microsoft.com/en-us/azure/container-apps/jobs?tabs=azure-cli#event-driven-jobs)

**Expected behavior** [What you expected to happen.]
Dedicated workload profile is scaling up from zero and down after job is completed.

**Actual behavior** [What actually happened.]
Job is not run, workload provide is not scaled.

**Screenshots**         
In Environment logs, KEDA seems to be logging events every 5 minutes
![image](https://github.com/microsoft/azure-container-apps/assets/2273250/8aab9bd3-0306-4fb1-8835-cceb943efcfa)

**Comment by nosalan** [comment]
To add more, this seems to not be an issue with scaling of workload profile but with Keda scaler.
The detailed scenario was:

- 2 jobs were running, the jobs are used for Azure Devops pipelines,
- for some reason Azure Devops got disconnected from jobs with the following error: `##[error]We stopped hearing from agent caj-XXXXXX-k95qm-jqtps. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610`
- while on Azure Devops part the jobs were abandoned, they were still in the Running state looking at the Container Apps Job resource. They had to be stopped manually with the `az containerapp job stop` command.
- At this point, I suspect that Keda scaler didn't catch that the jobs were stopped. This is because when 2 new Azure Devops jobs were run, they got stuck in a queue and Keda didn't trigger new jobs.
- When 3rd Azure Devops job was run, then Keda kicked in and created 1 new job.

In summary there seems to be 2 problems here:
1. Azure Devops pipeline got disconnected from Azure Container Apps Job with the error mentioned above,
2. Event-driven scaling gets out of sync - it takes into account jobs there were stopped.

**Comment by vinisoto** [comment]
do you see any errors in the console logs of the Jobs? What were your `replica-retry-limit` and `replica-timeout` values for your Job? 

Even if the scaler determines that it needs to scale down it will still honor those two settings

cc: @anthonychu

**Comment by nosalan** [comment]
Hi @vinisoto 
Following is set in bicep:
`replicaRetryLimit` = 1
`replicaTimeout` = 2400 (40 minutes)
`replicaCompletionCount` = 1
`parallelism` = 1

There were no logs from jobs at this time, the `ContainerAppConsoleLogs_CL` was empty, the jobs were stopped manually as mentioned in my previous comment. My guess was that the scaler wrongly thought that they were running still.

**Comment by kalprajkakani** [comment]
I'm experiencing the same issue. I manually stopped the job for some reason, and now event scaling is not working, while manual scaling is working fine. KEDA is logging "Created 0 jobs.". I have to delete and recreate the job to start event scaling again.

**Comment by vinisoto** [comment]
@nosalan - we recently fixed an issue in which some pods were not 'cleaned up' after they completed. This would cause the scaler to think that work was still being performed and there was no need for scaling. The fix for this is already rolled out world-wide. Can you please check if this fixes your scenario?

@kalprajkakani - We recently fixed a scenario in which manually stopping a Job execution prevents event-triggered Jobs to execute going forward (there is no mitigation other than deleting your Job and re-creating). The fix for this will begin rollout on the week of 11/27

**Comment by nosalan** [comment]
@vinisoto Thanks, I have not seen this problem recently.

**Comment by jomach** [comment]
I have the same issue now. Have a message in the queue and the container does not start. There are no logs on Monitoring page for this. how to debug it ?
